extensions/smallrye-health/spi/src/main/java/io/quarkus/smallrye/health/deployment/spi/HealthBuildItem.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.smallrye.health.deployment.spi;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+public final class HealthBuildItem extends MultiBuildItem {
+    private Class<?> healthCheckClass;
+
+    public HealthBuildItem(Class<?> healthCheckClass) {
```

Comment:
I think a health builditem could have a name: e.g. `quarkus.agroal.database_connection` (or something else). this is how we could do selective filtering down the line

Reply:
As the name is not currently used I didn't incude it already so we will nee to ask for it when filtering will be implemented.

Reply:
I was about to have it now as this will be a breaking change for extensions and wanted to avoid that but if we are unstable on the name concept, that's premature.

Reply:
I was asking myself the same: why not to add it now. 
But I don't think breaking change on extension is really an issue as we can revisit all extension that uses it at the same time. It's an internal breaking change that didn't impact users so it's OK.